### PR TITLE
Add suggestion for "ignore" doc code block

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/rust-lang/rust/issues/30032.

This PR adds a suggestion to help users when they have a "ignore" doc code block which is invalid rust code.

r? @jyn514

### CODE REVIEW

File: src/librustdoc/passes/check_code_block_syntax.rs [hunk 1]
```diff
@@ -67,6 +67,15 @@ impl<'a, 'tcx> SyntaxChecker<'a, 'tcx> {
                         String::from("```text"),
                         Applicability::MachineApplicable,
                     );
+                } else if suggest_using_text && code_block.is_ignore {
+                    let sp = sp.from_inner(InnerSpan::new(0, 3));
+                    diag.span_suggestion(
+                        sp,
+                        "`ignore` code blocks require valid Rust code. Mark blocks that do not \
```
Comment (jyn514) [inline src/librustdoc/passes/check_code_block_syntax.rs:74]:
```suggestion
                        "`ignore` code blocks require valid Rust code for syntax highlighting. Mark blocks that do not \
```

**Comment by GuillaumeGomez** [comment]
Updated!

File: src/librustdoc/html/markdown.rs [hunk 2]
```diff
@@ -1259,14 +1261,15 @@ crate fn rust_code_blocks(md: &str, extra_info: &ExtraInfo<'_, '_>) -> Vec<RustC
...
                                 syntax,
+                                is_ignore: lang_string.ignore != Ignore::None,
                             });
                             continue;
                         }
                     };
                     while let Some((Event::Text(_), offset)) = p.next() {
                         code_end = offset.end;
                     }
-                    (syntax, code_start, code_end, offset, true)
```
Comment (jyn514) [inline src/librustdoc/html/markdown.rs:1269]: You have `lang_string.ignore != Ignore::None` a lot - maybe make it a separate variable?

**Review by jyn514 — APPROVED** [review]
r=me with or without fixing the nit

**Pull request merged**